src/pyfmi/fmi3.pyx

```diff
@@ -886,6 +1111,174 @@ cdef class FMUModelME3(FMUModelBase3):
+            self._fmu,
+            noSetFMUStatePriorToCurrentPoint,
+            enterEventMode,
+            terminateSimulation
+        )
+        self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+
+        self._last_accepted_time = self._get_time()
+
+        if status != FMIL3.fmi3_status_ok:
+            raise FMUException('Failed to call FMI completed integrator step at time: %E.' % self.time)
+
+        return enterEventMode[0] == FMIL3.fmi3_true, terminateSimulation[0] == FMIL3.fmi3_true
+
+    cdef int _get_continuous_states_fmil(self, FMIL3.fmi3_float64_t[:] ndx):
+        cdef int status
+        if self._nContinuousStates > 0:
+            self._log_handler.capi_start_callback(self._max_log_size_msg_sent, self._current_log_size)
+            status = FMIL3.fmi3_import_get_continuous_states(self._fmu, &ndx[0] ,self._nContinuousStates)
+            self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+            return status
+        else:
+            return FMIL3.fmi3_status_ok
+
+    def _get_continuous_states(self):
+        """ Returns a vector with the values of the continuous states.
+
+            Returns::
+                The continuous states.
+        """
+        cdef int status
+        cdef np.ndarray[FMIL3.fmi3_float64_t, ndim=1, mode='c'] ndx = np.zeros(
+            self._nContinuousStates,
+            dtype = np.double
+        )
+
+        status = self._get_continuous_states_fmil(ndx)
+
+        if status != 0:
+            raise FMUException('Failed to retrieve the continuous states.')
+
+        return ndx
+
+    cdef int _set_continuous_states_fmil(self, FMIL3.fmi3_float64_t[:] ndx):
+        cdef int status
+        if self._nContinuousStates > 0:
+            self._log_handler.capi_start_callback(self._max_log_size_msg_sent, self._current_log_size)
+            status = FMIL3.fmi3_import_set_continuous_states(self._fmu, &ndx[0], self._nContinuousStates)
+            self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+            return status
+        else:
+            return FMIL3.fmi3_status_ok
+
+    def _set_continuous_states(self, np.ndarray[FMIL3.fmi3_float64_t, ndim=1, mode="c"] values):
+        """ Set the values of the continuous states.
+
+            Parameters::
+                values--
+                    The new values of the continuous states.
+        """
+        cdef int status
+        cdef np.ndarray[FMIL3.fmi3_float64_t, ndim=1,mode='c'] ndx = values
+
+        if np.size(ndx) != self._nContinuousStates:
+            raise FMUException(
+                'Failed to set the new continuous states. ' \
+                'The number of values are not consistent with the number of '\
+                f'continuous states, which are {self._nContinuousStates}.')
+
+        status = self._set_continuous_states_fmil(ndx)
+
+        if status >= 3:
```

Comment:
Use enum instead of 3?

Reply:
I changed this for now to `FMIL3.fmi3_status_t`, I think in general we should have a separate effort to return enums for all of the functions related to the `FMI3_*`, I would prefer not doing it in this ticket though (or maybe in the end) because of the reason I explained in the other comment about Cython unable to convert Enum instances between the Cython layer and Python.